doc/database-backup-and-restore.md

```diff
@@ -0,0 +1,106 @@
+
+1. Sign into the platform from your machine
+    ```
+    cf login
+    ```
+1. Install the local [Conduit plugin](https://github.com/alphagov/paas-cf-conduit) for Cloudfoundry. This step warns of untrusted packages. We trust that the author is the Government Digital Service and confirm
+    ```
+    cf install-plugin conduit
+    ```
+1. Create a new backup of production
+    ```
+    cf target -s prod
+    cf services # to get the SERVICE_NAME of postgres
+    cf conduit SERVICE_NAME -- pg_dump --file PROD_DATA_FILE_NAME.sql
+    ```
+1. Create a backup of the database we are about to overwrite and keep this as a local file
+    ```
+    cf target -s staging
+    cf services # to get the SERVICE_NAME
+    cf conduit SERVICE_NAME -- pg_dump --file STAGING_DATA_FILE_NAME.sql
+    ```
+1. Find the Database role name from the database dump. It should be of the form: `rdsbroker_<uuid>_manager`
+    ```
+    grep -n rdsbroker PROD_DATA_FILE_NAME.sql
+    ```
+1. Connect to the target database with a PostgreSQL prompt
+    ```
+    cf target -s staging
+    cf services # to get the SERVICE_NAME for the destination postgres
+    cf conduit SERVICE_NAME -- psql
+    ```
+1. Wipe the target staging database. Inserting the role name from the PROD_DATA_FILE_NAME.sql
+    ```
+    DROP DATABASE "roda-staging";
+    CREATE DATABASE "roda-staging";
+    CREATE ROLE "rdsbroker_<uuid>_manager" WITH SUPERUSER CREATEDB CREATEROLE;
+    ```
+1. Add data to the target staging database
+    ```
+    cf target -s staging
+    cf services # to get the DESTINATION_SERVICE_NAME for the destination postgres
+    cf conduit DESTINATION_SERVICE_NAME -- psql -d roda-staging < PROD_DATA_FILE_NAME.sql
+    ```
+1. Ask Rake to prepare the database [using a rails console](/doc/console-access.md)
+    ```
+    rake db:prepare
+    ```
+1. Seed the database with generic roda@dxw.com and roda+dp@dxw.com users to allow us to sign in. [Use a rails console](/doc/console-access.md) to add the generic users by copying the contents of the `db/seeds/staging_users.rb`
+1. Sign into the service to verify data is present
+
+### PaaS to local
+
+In this example we will overwrite our local development database with the contents of the production database.
+
+---
+
+1. Create a new backup of production
+    ```
+    cf install-plugin conduit
+    cf target -s prod
+    cf services # to get the SERVICE_NAME of postgres
+    cf conduit SERVICE_NAME -- pg_dump --file PROD_DATA_FILE_NAME.sql
+    ```
+1. Find the Database role name from the database dump. It should be of the form: `rdsbroker_<uuid>_manager`
+    ```
+    grep -n rdsbroker PROD_DATA_FILE_NAME.sql
+    ```
+1. Use the Postgres binary to wipe and prepare the local database (if you've done this process before you'll get a warning that this role already exists)
+    ```
+    psql
```

Comment:
When devs have installed postgres with Homebrew this will often need to be `psql postgres` - don't think we need to do anything, just something I've seen.